packages/grid_client/tests/modules/gateways.test.ts

```diff
@@ -231,6 +234,295 @@ test("TC1237 - Gateways: Expose a VM Over Gateway", async () => {
   } else {
     throw new Error("Gateway is unreachable after multiple retries");
   }
+
+  //--------------------Mycelium--------------------
+  backends = ["http://[" + result[0].myceliumIP + "]:8000"];
+  log(backends);
+
+  //Name Gateway Model
+  const gw1: GatewayNameModel = {
+    name: name + "c",
+    node_id: gatewayNodeId,
+    tls_passthrough: tlsPassthrough,
+    backends: backends,
+  };
+
+  gatewayRes = await gridClient.gateway.deploy_name(gw1);
+  log(gatewayRes);
+
+  //Contracts Assertions
+  expect(gatewayRes.contracts.created).toHaveLength(1);
+  expect(gatewayRes.contracts.updated).toHaveLength(0);
+  expect(gatewayRes.contracts.deleted).toHaveLength(0);
+  expect(gatewayRes.contracts.created[0].contractType.nodeContract.nodeId).toBe(gatewayNodeId);
+
+  gatewayResult = await gridClient.gateway.getObj(gw1.name);
+  log(gatewayResult);
+
+  //Gateway Assertions
+  expect(gatewayResult[0].name).toBe(name + "c");
+  expect(gatewayResult[0].status).toBe("ok");
+  expect(gatewayResult[0].type).toContain("name");
+  expect(gatewayResult[0].domain).toContain(name);
+  expect(gatewayResult[0].tls_passthrough).toBe(tlsPassthrough);
+  expect(gatewayResult[0].backends).toStrictEqual(backends);
+
+  domain = "https://" + gatewayResult[0].domain;
+  reachable = false;
+
+  for (let i = 0; i < 30; i++) {
+    axios
+      .get(domain)
+      .then(() => {
+        log("gateway is reachable");
+        reachable = true;
+      })
+      .catch(() => {
+        log("gateway is not reachable");
+      });
+    if (reachable) {
+      break;
+    }
+    const wait = await setTimeout(5000, "Waiting for gateway to be ready");
+    log(wait);
+  }
```

Comment:
Move this section to another function e.g. `reachGateway`

Reply:
Good point!